### ISSUE TITLE
Editor crash in UndoRedo::_process_operation_list when moving resource in FileSystemDock

### ISSUE THREAD

**Issue by DrMoriarty** [godotengine/godot#28528]
**Godot version:**

Godot 3.1.1.dev (4764e1797)
Godot 3.1.1.official
Godot 3.1.2-devel (e802da4e8)

**OS/device including version:**

MacOSX 10.13.6

**Issue description:**

I move some png resource from one folder to another. 

Godot 3.1.dev (4764e1797) shows this crashlog:
```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007fff7686af5a _sigtramp + 26
[2] 2   ???                                 0x000000011631fa50 0x0 + 4667341392
[3] String::size() const (in godot.osx.tools.64) (ustring.h:151)
[4] String::length() const (in godot.osx.tools.64) (ustring.h:194)
[5] String::operator==(String const&) const (in godot.osx.tools.64) (ustring.cpp:201)
[6] StringName::StringName(String const&) (in godot.osx.tools.64) (string_name.cpp:281)
[7] StringName::StringName(String const&) (in godot.osx.tools.64) (string_name.cpp:321)
[8] UndoRedo::_process_operation_list(List<UndoRedo::Operation, DefaultAllocator>::Element*) (in godot.osx.tools.64) (undo_redo.cpp:301)
[9] UndoRedo::redo() (in godot.osx.tools.64) (undo_redo.cpp:333)
[10] UndoRedo::commit_action() (in godot.osx.tools.64) (undo_redo.cpp:255)
[11] EditorNode::_scene_tab_changed(int) (in godot.osx.tools.64) (editor_node.cpp:0)
[12] EditorNode::reload_scene(String const&) (in godot.osx.tools.64) (editor_node.cpp:4635)
[13] FileSystemDock::_update_dependencies_after_move(Map<String, String, Comparator<String>, DefaultAllocator> const&) const (in godot.osx.tools.64) (filesystem_dock.cpp:1112)
[14] FileSystemDock::_move_operation_confirm(String const&, bool) (in godot.osx.tools.64) (filesystem_dock.cpp:1337)
[15] MethodBind2<String const&, bool>::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:1428)
[16] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:942)
[17] Object::emit_signal(StringName const&, Variant const**, int) (in godot.osx.tools.64) (object.cpp:1228)
[18] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (object.cpp:1284)
[19] EditorDirDialog::ok_pressed() (in godot.osx.tools.64) (editor_dir_dialog.cpp:128)
[20] AcceptDialog::_ok_pressed() (in godot.osx.tools.64) (dialogs.cpp:387)
[21] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:56)
[22] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:942)
[23] Object::emit_signal(StringName const&, Variant const**, int) (in godot.osx.tools.64) (object.cpp:1228)
[24] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (object.cpp:1284)
[25] Tree::_gui_input(Ref<InputEvent>) (in godot.osx.tools.64) (tree.cpp:2643)
[26] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:729)
[27] Object::call_multilevel(StringName const&, Variant const**, int) (in godot.osx.tools.64) (object.cpp:777)
[28] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (object.cpp:882)
[29] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (in godot.osx.tools.64) (viewport.cpp:1535)
[30] Viewport::_gui_input_event(Ref<InputEvent>) (in godot.osx.tools.64) (viewport.cpp:1837)
[31] Viewport::input(Ref<InputEvent> const&) (in godot.osx.tools.64) (viewport.cpp:2661)
[32] Viewport::_vp_input(Ref<InputEvent> const&) (in godot.osx.tools.64) (viewport.cpp:1312)
[33] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:729)
[34] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:942)
[35] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (object.cpp:867)
[36] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (scene_tree.cpp:262)
[37] SceneTree::input_event(Ref<InputEvent> const&) (in godot.osx.tools.64) (scene_tree.cpp:418)
[38] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (in godot.osx.tools.64) (input_default.cpp:442)
[39] InputDefault::parse_input_event(Ref<InputEvent> const&) (in godot.osx.tools.64) (input_default.cpp:260)
[40] InputDefault::flush_accumulated_events() (in godot.osx.tools.64) (input_default.cpp:680)
[41] OS_OSX::process_events() (in godot.osx.tools.64) (os_osx.mm:2575)
[42] OS_OSX::run() (in godot.osx.tools.64) (os_osx.mm:2651)
[43] main (in godot.osx.tools.64) (godot_main_osx.mm:100)
[44] 44  libdyld.dylib                       0x00007fff7655c015 start + 1
[45] 45  ???                                 0x0000000000000004 0x0 + 4
-- END OF BACKTRACE --
```

Godot 3.1.1.official shows this crashlog:
```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007fff7686af5a _sigtramp + 26
[2] 2   ???                                 0x00007fa5dcf57b00 0x0 + 140350353406720
[3] StringName::StringName(String const&) (in Godot) + 73
[4] UndoRedo::_process_operation_list(List<UndoRedo::Operation, DefaultAllocator>::Element*) (in Godot) + 1164
[5] UndoRedo::redo() (in Godot) + 170
[6] UndoRedo::commit_action() (in Godot) + 57
[7] EditorNode::_scene_tab_changed(int) (in Godot) + 1739
[8] EditorNode::reload_scene(String const&) (in Godot) + 962
[9] FileSystemDock::_update_dependencies_after_move(Map<String, String, Comparator<String>, DefaultAllocator> const&) const (in Godot) + 886
[10] FileSystemDock::_move_operation_confirm(String const&, bool) (in Godot) + 830
[11] MethodBind2<String const&, bool>::call(Object*, Variant const**, int, Variant::CallError&) (in Godot) + 454
[12] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in Godot) + 355
[13] Object::emit_signal(StringName const&, Variant const**, int) (in Godot) + 2016
[14] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in Godot) + 114
[15] EditorDirDialog::ok_pressed() (in Godot) + 207
[16] AcceptDialog::_ok_pressed() (in Godot) + 60
[17] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (in Godot) + 98
[18] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in Godot) + 355
[19] Object::emit_signal(StringName const&, Variant const**, int) (in Godot) + 2016
[20] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in Godot) + 114
[21] Tree::_gui_input(Ref<InputEvent>) (in Godot) + 8697
[22] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (in Godot) + 304
[23] Object::call_multilevel(StringName const&, Variant const**, int) (in Godot) + 232
[24] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in Godot) + 115
[25] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (in Godot) + 756
[26] Viewport::_gui_input_event(Ref<InputEvent>) (in Godot) + 2200
[27] Viewport::input(Ref<InputEvent> const&) (in Godot) + 294
[28] Viewport::_vp_input(Ref<InputEvent> const&) (in Godot) + 234
[29] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (in Godot) + 304
[30] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in Godot) + 355
[31] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in Godot) + 136
[32] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in Godot) + 1337
[33] SceneTree::input_event(Ref<InputEvent> const&) (in Godot) + 384
[34] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (in Godot) + 3165
[35] InputDefault::flush_accumulated_events() (in Godot) + 41
[36] OS_OSX::run() (in Godot) + 614
[37] main (in Godot) + 811
[38] 38  libdyld.dylib                       0x00007fff7655c015 start + 1
-- END OF BACKTRACE --
```

Godot 3.1.2-devel (e802da4e8) shows this crashlog:
```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007fff7686af5a _sigtramp + 26
[2] 2   ???                                 0x000000010f9d8a38 0x0 + 4556950072
[3] String::size() const (in godot.osx.tools.64) (ustring.h:151)
[4] String::length() const (in godot.osx.tools.64) (ustring.h:194)
[5] String::operator==(String const&) const (in godot.osx.tools.64) (ustring.cpp:201)
[6] StringName::StringName(String const&) (in godot.osx.tools.64) (string_name.cpp:281)
[7] StringName::StringName(String const&) (in godot.osx.tools.64) (string_name.cpp:321)
[8] UndoRedo::_process_operation_list(List<UndoRedo::Operation, DefaultAllocator>::Element*) (in godot.osx.tools.64) (undo_redo.cpp:301)
[9] UndoRedo::redo() (in godot.osx.tools.64) (undo_redo.cpp:333)
[10] UndoRedo::commit_action() (in godot.osx.tools.64) (undo_redo.cpp:255)
[11] EditorNode::_scene_tab_changed(int) (in godot.osx.tools.64) (editor_node.cpp:0)
[12] EditorNode::reload_scene(String const&) (in godot.osx.tools.64) (editor_node.cpp:4635)
[13] FileSystemDock::_update_dependencies_after_move(Map<String, String, Comparator<String>, DefaultAllocator> const&) const (in godot.osx.tools.64) (filesystem_dock.cpp:1112)
[14] FileSystemDock::_move_operation_confirm(String const&, bool) (in godot.osx.tools.64) (filesystem_dock.cpp:1337)
[15] MethodBind2<String const&, bool>::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:1428)
[16] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:942)
[17] Object::emit_signal(StringName const&, Variant const**, int) (in godot.osx.tools.64) (object.cpp:1228)
[18] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (object.cpp:1284)
[19] EditorDirDialog::ok_pressed() (in godot.osx.tools.64) (editor_dir_dialog.cpp:128)
[20] AcceptDialog::_ok_pressed() (in godot.osx.tools.64) (dialogs.cpp:387)
[21] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:56)
[22] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:942)
[23] Object::emit_signal(StringName const&, Variant const**, int) (in godot.osx.tools.64) (object.cpp:1228)
[24] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (object.cpp:1284)
[25] Tree::_gui_input(Ref<InputEvent>) (in godot.osx.tools.64) (tree.cpp:2643)
[26] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:729)
[27] Object::call_multilevel(StringName const&, Variant const**, int) (in godot.osx.tools.64) (object.cpp:777)
[28] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (object.cpp:882)
[29] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (in godot.osx.tools.64) (viewport.cpp:1535)
[30] Viewport::_gui_input_event(Ref<InputEvent>) (in godot.osx.tools.64) (viewport.cpp:1837)
[31] Viewport::input(Ref<InputEvent> const&) (in godot.osx.tools.64) (viewport.cpp:2661)
[32] Viewport::_vp_input(Ref<InputEvent> const&) (in godot.osx.tools.64) (viewport.cpp:1312)
[33] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:729)
[34] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:942)
[35] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (object.cpp:867)
[36] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (in godot.osx.tools.64) (scene_tree.cpp:262)
[37] SceneTree::input_event(Ref<InputEvent> const&) (in godot.osx.tools.64) (scene_tree.cpp:418)
[38] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (in godot.osx.tools.64) (input_default.cpp:442)
[39] InputDefault::parse_input_event(Ref<InputEvent> const&) (in godot.osx.tools.64) (input_default.cpp:260)
[40] InputDefault::flush_accumulated_events() (in godot.osx.tools.64) (input_default.cpp:680)
[41] OS_OSX::process_events() (in godot.osx.tools.64) (os_osx.mm:2575)
[42] OS_OSX::run() (in godot.osx.tools.64) (os_osx.mm:2651)
[43] main (in godot.osx.tools.64) (godot_main_osx.mm:100)
[44] 44  libdyld.dylib                       0x00007fff7655c015 start + 1
[45] 45  ???                                 0x0000000000000004 0x0 + 4
-- END OF BACKTRACE --
```

**Steps to reproduce:**

Move png resource from one folder to another. Sometimes it doesn't crash after first moving, so second moving will do the trick.

**Minimal reproduction project:**

[Editor Crash.zip](https://github.com/godotengine/godot/files/3130376/Editor.Crash.zip)

The same project I uploaded to #28395

**Comment by DrMoriarty** [comment]
The last checking 3.1.2-devel on minimal reproduction project gave me the different crashlog:
```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] 1   libsystem_platform.dylib            0x00007fff7686af5a _sigtramp + 26
[2] 2   ???                                 0x00007f80b232fac0 0x0 + 140190722226880
[3] Node* Object::cast_to<Node>(Object*) (in godot.osx.tools.64) (object.h:588)
[4] Variant::operator Node*() const (in godot.osx.tools.64) (variant.cpp:1763)
[5] VariantObjectClassChecker<Node*>::check(Variant const&) (in godot.osx.tools.64) (method_bind.h:143)
[6] MethodBind2<Dictionary, Node*>::call(Object*, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (method_bind.gen.inc:1425)
[7] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (in godot.osx.tools.64) (object.cpp:942)
[8] MessageQueue::_call_function(Object*, StringName const&, Variant const*, int, bool) (in godot.osx.tools.64) (message_queue.cpp:0)
[9] MessageQueue::flush() (in godot.osx.tools.64) (message_queue.cpp:305)
[10] SceneTree::iteration(float) (in godot.osx.tools.64) (scene_tree.cpp:477)
[11] Main::iteration() (in godot.osx.tools.64) (main.cpp:1862)
[12] OS_OSX::run() (in godot.osx.tools.64) (os_osx.mm:2654)
[13] main (in godot.osx.tools.64) (godot_main_osx.mm:100)
[14] 14  libdyld.dylib                       0x00007fff7655c015 start + 1
[15] 15  ???                                 0x0000000000000004 0x0 + 4
-- END OF BACKTRACE --
```

**Comment by hpvb** [comment]
This was probably fixed in #29459 can you retest this on MacOS please?

**Comment by DrMoriarty** [comment]
@hpvb I just checked on MacOSX 10.14.4 (build from commit 197b65f32ac811f79bc5599fbfe8cf83914b6873) 
Godot doesn't crash when I move png resources. Also it seems that this bug already fixed too: #28395

**Comment by hpvb** [comment]
Thanks for testing!
